### PR TITLE
wxWidges and PCM

### PR DESCRIPTION
**Pull request by leoheck**
The latest 2 PRs were combined in this one

### This comes from the PR 2 (wxWidgets)

Replaces GUI (TkInter -> wxWidgets)

It looks like it was easier than I could possibly imagine. I could make a working GUI with wx.
I used wxGlade tool to generate a similar GUI that Kidiff had before. It generates a python file that I am not touching and then I am using another python file to tweak it a bit with settings I could not find on wxGlade. I am pretty happy with the result, at least on my linux box.

On mac it does not look pretty yet, but it can be improved further, this is just the initial result.

image

It requires wx on macOS, I could install it with
```brew install wxpython```

### This comes from the PR 3 (PCM)

This PR adds initial support for KiDiff landing on the Kicad toolbar through PCM

The icon is currently the rightmost (red/blue)
And it launches the new main window made with wxWidgets

There is a script to install it manually. Useful to test, and while this is not added on Kicad's Git lab. But first, it has to be merged here, and then I will create a new tag to make the integration happen.

image

### New stuff

- README was updated
- pygubu (Tk designer) was removed from requirements.txt 
- `env-nightly.sh` removed

### CODE REVIEW

**Comment by leoheck** [comment]
This is the comment you made before. I still have to investigate what is happening and what you have said, actually. I may not have understood this comment. But saving it here. 

https://github.com/Gasman2014/KiCad-Diff/pull/78#issuecomment-1020999900

**Comment by leoheck** [comment]
I took the liberty to replace the images in the README to have something revamped with fewer images. I hope you don't mind with this change.

**Comment by leoheck** [comment]
I am going to improve this Commits selection GUI when I have the MacBook with me. It does not look good at all.

**Comment by leoheck** [comment]
I improved the organization of the project a bit. 
- Python-related files are in the `kidiff` folder
- Bash scripts are in the bin folder now.

**Comment by Gasman2014** [comment]
Still hitting the 
```This program needs access to the screen. Please run with a
Framework build of python, and only when you are logged in
on the main display of your Mac```
issue and have made no progress with this. Any luck with your testing? I could roll another user to see if its too many years of cruft on my system - so I have not been able to test any of this, I'm afraid. I've exhausted Google ...

**Comment by leoheck** [comment]
Ah, sure I paused this a bit I was not feeling good, and now I lost the chance to use my wife's MacBook. Now it will take a bit longer until I can use it again.

**Comment by leoheck** [comment]
@Gasman2014 

Hey John, a user was complaining that Kiri was presenting vias on all layers even if they were buried/blinded, and since I never done any design with those kind of vias, I never saw this issue myself.

This is maybe something related to what you said long time ago when you mentioned vias were weird.

I found this setting here that disables drill marks, which makes the plotted svgs behave better regarding the visibility of the vias. 

```python
popt.SetDrillMarksType(pcbnew.PCB_PLOT_PARAMS.NO_DRILL_SHAPE)
````

In other words... we don't want to see the "holes" on every layer but the annular rings of the vias when they are in use.

**Comment by leoheck** [comment]
Oh, that was unexpected. Could you use wxWidgets version of the tool?

**Comment by Gasman2014** [comment]
Sorry not been able to test but keen to tie up loose ends and clear the decks.

Best Wishes

John

**Comment by leoheck** [comment]
Not a problem at all, I am using my fork of it when I need it, so not a problem at all. But I am glad you merged it.
